src/datasets/processors/subsample/script.py

```diff
@@ -56,31 +71,76 @@
     choice_probs = [ probs[batch] for batch in choice_batch ]
     obs_index = np.random.choice(choice_ix, size=n_obs, replace=False, p=choice_probs)
 else:
-    obs_index = np.random.choice(np.where(obs_filt)[0], n_vars, replace=False)
+    obs_index = np.random.choice(np.where(obs_filt)[0], n_obs, replace=False)
+
+# subsample obs
+adata_output = adata_input[obs_index].copy()
+if par["input_mod2"] is not None:
+    adata_output_mod2 = adata_mod2[obs_index].copy()
+
+# filter cells and genes
+if par["input_mod2"] is not None:
+    n_cells =  scprep.utils.toarray(adata_output.X.sum(axis=1)).flatten()
+    n_cells_mod2 =  scprep.utils.toarray(adata_output_mod2.X.sum(axis=1)).flatten()
+    keep_cells = np.minimum(n_cells, n_cells_mod2) > 1
+    adata_output = adata_output[keep_cells, :].copy()
+    adata_output_mod2 = adata_output_mod2[keep_cells, :].copy()
+
+    sc.pp.filter_genes(adata_output, min_cells=1)
+    sc.pp.filter_genes(adata_output_mod2, min_cells=1)
+    
+else:
+    # todo: this should not remove features in keep_features!
+    print(">> Remove empty observations and features", flush=True)
+    sc.pp.filter_genes(adata_output, min_cells=1)
+    sc.pp.filter_cells(adata_output, min_counts=2)
 
 print(">> Subsampling the features", flush=True)
 if par.get("keep_features"):
-    initial_filt = adata_input.var_names.isin(par["keep_features"])
+    initial_filt = adata_output.var_names.isin(par["keep_features"])
     initial_idx, *_ = initial_filt.nonzero()
     remaining_idx, *_ = (~initial_filt).nonzero()
     rest_idx = remaining_idx[np.random.choice(len(remaining_idx), n_vars - len(initial_idx), replace=False)]
     var_ix = np.concatenate([initial_idx, rest_idx])
 else:
-    var_ix = np.random.choice(adata_input.shape[1], n_vars, replace=False)
-
-
-adata_output = adata_input[obs_index, var_ix].copy()
+    var_ix = np.random.choice(adata_output.shape[1], n_vars, replace=False)
+    if par["input_mod2"] is not None:
+        var_ix_mod2 = np.random.choice(adata_output_mod2.shape[1], n_vars, replace=False)
+
+#  subsample vars
+adata_output = adata_output[:, var_ix].copy()
+if par["input_mod2"] is not None:
+    adata_output_mod2 = adata_output_mod2[:, var_ix_mod2].copy()
+
+# filter cells and genes
+if par["input_mod2"] is not None:
+    n_cells =  scprep.utils.toarray(adata_output.X.sum(axis=1)).flatten()
+    n_cells_mod2 =  scprep.utils.toarray(adata_output_mod2.X.sum(axis=1)).flatten()
```

Comment:
Likewise

It seems a bit redundant to install scprep just for this.

```suggestion
    n_cells =  adata_output.X.sum(axis=1).A.flatten()
    n_cells_mod2 =  adata_output_mod2.X.sum(axis=1).A.flatten()
```